### ISSUE TITLE
Change name of tumbler

### ISSUE THREAD

**Issue by chris-belcher** [chris-belcher/joinmarket#238]
Turns out the word tumbler used to refer to improving bitcoin privacy dates back to the old silk road marketplace. This is not really the image we want to show.

Tumbler.py needs a new name. Something which doesn't allude to money laundering or any other criminal activity.

For example if someone's bitcoins get rejected with the message *"Your coins were used for criminal activity X transactions ago and are therefore tained"* they need to know that this bot can solve their problem.

Ideas: privacy-o-matic.py, multijoin.py, unlink-coins.py, multiple-coinjoin.py

**Comment by JulianTosh** [comment]
In the bitcoin classes I teach (including law enforcement), I use the term "Dry cleaning" when mixing for privacy, as opposed to "money laundering" to hide the source of funds from illicit activity.

I vote for drycleaner.py!

**Comment by HostFat** [comment]
A simpler "cleaner.py" can even be good :)

**Comment by chris-belcher** [comment]
That's not a bad idea, or even coin-cleaner.py to allude to coin collectors polishing their stack.

**Comment by JulianTosh** [comment]
How about fung.py as in fungible?

- Sent from a superior 9.5 cubic inch Android mobile computing device.

**Comment by chris-belcher** [comment]
Few people know what fungibility is, it seems too academic I think.

**Comment by chris-belcher** [comment]
The name cleaner.py could also allude to money laundering, although not very much.

**Comment by tailsjoin** [comment]
untrace.py

disorder.py

entropy.py

disarray.py

crumble.py

stumble.py

**Comment by chris-belcher** [comment]
untracer.py is good out of those, describes what the script does.

Maybe for the some of the others, transform the name into a verb.

disorderer.py
entropy-increaser.py
disarrayer.py

**Comment by JulianTosh** [comment]
One more shot at fungibility angle...

Fungi.py (fun-jih-pie)

A pythonic play on fungify.py

- Sent from a superior 9.5 cubic inch Android mobile computing device.

**Comment by chris-belcher** [comment]
Even so @JulianTosh if someone comes to JoinMarket wondering how to do a tumble of their coins, they will probably ask themselves what on earth fungibility and fungi.py is, those words are hardly ever used outside of finance and economics.

**Comment by JulianTosh** [comment]
So what is the point of changing the name? Is it is to get away from a
negative image like money laundering? Then what better way than to teach
people what they should know, understand, and for the real, underlying
premise? Should they never learn what fungibility means because they don't
already?

Mixing bitcoins results in the very definition of fungibility.

- Sent from a superior 9.5 cubic inch Android mobile computing device.

**Comment by HostFat** [comment]
I also think that untracer.py is good and easy to understand :)

**Comment by chris-belcher** [comment]
@JulianTosh Yes, the point is to rebrand it. Teaching people is counterproductive, people are lazy and time-stressed. They just need to get their job done and not learn about

**Comment by adlai** [comment]
untaint.py

**Comment by AdamISZ** [comment]
Since we're all having fun here I'll resubmit my suggestion of 'taint-thinner.py' :)

**Comment by tailsjoin** [comment]
taint-cleaner.py ;)

**Comment by chris-belcher** [comment]
I don't think the word taint is good.
The word 'taint' comes from blockchain.info's explorer, it's never been entirely clear whether taint means anything at all. Now tools are more likely to look at the closure, as in this tool http://nullrefer.com/?https://www.elliptic.co/intelligence/ or they may correlate amount sizes.

Also people who are new to bitcoin may have no idea what taint means in this context.

What about allusions to cleaning against disease? disinfect.py, purify.py, sterlize.py

Or even allusions to historical revisionism: forget.py, fresh-start.py, disremember.py forgive.py, pardon.py

Right now I'm thinking untracer.py cleaner.py or with coin- added are the best.

**Comment by chris-belcher** [comment]
I've just learned that 'mixing' is a common terminology in crypto ,and not specific to tumbling

see https://en.wikipedia.org/wiki/Mix_network

so mixer.py ?

**Comment by JulianTosh** [comment]
mixer/ing still brings the idea of money laundering to mind.

since you are on the crypto route, what about

hasher.py
inputhasher.py
bithasher.py

- Sent from a superior 9.5 cubic inch Android mobile computing device.

**Comment by tailsjoin** [comment]
I like mixer.py. We should keep it simple. @JulianTosh I like your ideas, but I fear most people will not understand what it's use is without explanation. That explaining will become exhausting, and is unneeded. 

Honestly, I prefer tumbler or mixer to any other suggestion as people know what they are referring to. Regardless of their origin, we are still at a time where we can sway public perception of such things quite easily.

**Comment by tailsjoin** [comment]
Inspired by #244, maybe we should consider blender.py.

**Comment by K1773R** [comment]
mixer / blender is fine :)

**Comment by ghtdak** [comment]
During refactoring, I came across an apparently well known problem with naming.  I could describe it but, I think the following one-line in irc posted by <pigeons> says it all

```16:06 <pigeons> try-and-raise-yield-with-class-or-not.py ;)```

So, yield-generator.py is a double offender.  But we have multiple with '-' (interpreted as subtraction) and, yeah, I guess we want to avoid substrings in the example above.

I made the temporary executive decision of changing dashes to underscores and changing yield to btc but I'm definitely not happy with the latter.

**Comment by chris-belcher** [comment]
I'd caution against removing the word yield. It's a well-defined and commonly-understood word from investing/finance.
Even though yield is a reserved word in python, yield_generator is not.

If not yield, maybe try the word "growth", so bitcoin_grower.py or grow_bitcoins.py. Or return, as in return-on-capital, but then again return is a reserved word too.

**Comment by dergigi** [comment]
I realize that this discussion stopped in 2015, but I'd like to cast a vote for "scheduler" anyway.

All the tumbler script does—and yes, it is still called that—is schedule multiple coinjoins. The [code references "schedule" in many places](https://github.com/JoinMarket-Org/joinmarket-clientserver/search?q=schedule) and it is even referred to as "scheduler" in [one comment](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/52eeeb09b9a67e87ba59edac47ed7a2c22039df4/scripts/joinmarket-qt.py#L520).

Seven years on, and "tumbler" is still kinda bad for all the reasons outlined above.
